src/ghdcbot/core/social_validators.py

```
@@ -0,0 +1,259 @@
+            url_candidate = f"https://{url_candidate}"
+
+        if "://" in url_candidate:
+            parsed = urlparse(url_candidate)
+            host = (parsed.hostname or "").lower()
+            if host.startswith("www."):
+                host = host[4:]
+            if host in {"x.com", "twitter.com"}:
+                username = parsed.path.strip("/").split("/", 1)[0]
+                if username:
+                    return username
+                raise ValueError("Could not extract username from URL")
```

Ruff (0.15.18)

[warning] Avoid specifying long messages outside the exception class

(TRY003)